### DISCUSSION TITLE
Question: License

### DISCUSSION THREAD

**Discussion by zoeleu** [FunkinCrew/Funkin#1496] [Q&A] [answered]
## What is your question?
Why use Apache license, and not GPL license, to force more open source in the mods community? This would benefit Linux users to get source code and build mods such as Whitty or FPS Plus.

**Comment by c0mp3r** [comment]
I would like the license to change to GPL. It would force all of those mods only available for Windows to release their source code if they want to be up to date and allow Mac and Linux users to create binaries for their platforms, I don't understand the apache license.

#### TL;DR, mods for other platforms than Windows.

**Comment by ghost** [comment]
Wine on Linux surprisingly works for Windows fnf but must be in working directory, so I can do Windows mods

**Comment by c0mp3r** [comment]
Thanks for the sugestion. Will try to run mods with wine and see how it runs.

**Comment by ghost** [comment]
FPS counter is gone, sound is great, works very well, good performance.

**Comment by ghost** [comment]
You need to go into the folder with Funkin.exe before running Wine otherwise it is broken

**Comment by zoeleu** [comment]
Wine is a workaround. Not a solution. You don't know if any mods contain
malware in it's code.

Em dom., 2 de mai. de 2021 às 06:12, VMGuy23 ***@***.***>
escreveu:

**Comment by zoeleu** [comment]
@ninjamuffin99 your opinion on licenses?

**Comment by ninjamuffin99** [comment, marked as answer]
i mentioned it before, but I have been told by people that I trust that while GPL would be cool, it would probably cause issues down the line if we want to do business with people like console manufacturers, who often try to steer clear of GPL type stuff, since you need to have a lot of due diligence to make sure it's all in order and whatnot. Something like that. This isn't legal advice, but that's just the basic gist of it.
And also I think it'd get iffy. I want MODS to be open source, I don't really care if people take 4 lines of code for their own project. I think it's fair to have that freedom go both ways like that. I just REALLY HATE mods that are not open source.

**Reply by AbnormalPoof** [reply]
Wait so you hate all the multiplayer mods-

**Comment by zoeleu** [comment]
You should talk to a lawyer about these things, if it's possible to GPL on
consoles, and all the license problems. You can't change the license
suddenly without the approval of all contributors.

**Reply by ninjamuffin99** [reply]
from what I been told, that depends on the license. If I changed from Apache 2.0 to GPL, I can do that or whatev. But if people contribute to the code under GPL, THEN I would have to get explicit permission to change it back to something like Apache. In general though, I don't think GPL is what I want anyways. Like I mention, I really just care only for mods being sorta "force" open source. Otherwise apache meets most of the standards that I want out of FNF being open source as of right now

**Comment by zoeleu** [comment]
They need to be open source. That's it.

Em seg., 27 de set. de 2021 às 14:02, AbnormalPoof ***@***.***>
escreveu:
